src/views/catalog.vue

```diff
@@ -220,6 +209,11 @@ export default defineComponent({
       else this.prodCatalogCategoryTypeId = value
       this.getCatalogProducts()
     },
+    async viewProduct(product: any) {
+      await this.store.dispatch('product/addCatalogProductToCache', product).then(() => {
```

Comment:
We could skip this or we could use the cached variant by excluding it from fetched variants